### PR TITLE
[FIX] Send: better show input errors (RT-2975)

### PR DESCRIPTION
**Pull request by darkdarkdragon**
Re-validate input when currency changed,
and do not start pathfinding if input
is not valid

### CODE REVIEW

File: src/js/tabs/exchange.js [hunk 1]
```diff
@@ -54,9 +52,29 @@ ExchangeTab.prototype.angular = function (module)
         exchange.currency_name = currency.to_human({
           full_name: $scope.currencies_all_keyed[currency.get_iso()] ? $scope.currencies_all_keyed[currency.get_iso()].name : null
         });
-        $scope.update_exchange();
+        setImmediate(function() {
+          if ($scope.exchangeForm.amount !== undefined) {
+            $scope.$apply(function() {
+              // hack to re-validate input. remove this and uncomment $validate() when upgraded to angularjs 1.3
```
Comment (rht) [inline src/js/tabs/exchange.js]: It is already angularjs 1.3 https://github.com/ripple/ripple-client/blob/develop/bower.json#L6

File: src/js/tabs/exchange.js [hunk 2]
```diff
@@ -138,7 +158,11 @@ ExchangeTab.prototype.angular = function (module)
 
           var lastUpdate;
 
-          pf.on('update', function (upd) {
+          pf.on('update', function(upd) {
+            if (pf == null) {
```
Comment (jublo) [inline src/js/tabs/exchange.js]: Use `===` to compare with `null`.

File: test/unit/tabs/exchangeTabSpec.js [hunk 3]
```diff
@@ -45,12 +46,16 @@ describe('ExchangeCtrl', function() {
         account: 'r4EwBWxrx5HxYRyisfGzMto3AT8FZiYdWk'
       }
     };
+    */
```
Comment (jublo) [inline test/unit/tabs/exchangeTabSpec.js]: Any particular reason why to keep the old code as block comment?

Comment (darkdarkdragon) [inline test/unit/tabs/exchangeTabSpec.js]: no, just forgot to cleanup

File: src/js/tabs/exchange.js (hunk 2 above)
Comment (darkdarkdragon) [inline src/js/tabs/exchange.js]: in this case pf was undefined at start, so can't use ===

Comment (jublo) [inline src/js/tabs/exchange.js]: Where is the `var pf` declaration, then?

**Comment by mrajvanshy** [comment]
@darkdarkdragon rebase

**Comment by vhpoet** [comment]
LGTM

**Comment by mrajvanshy** [comment]
LGTM

**Pull request merged**